### PR TITLE
Allow dependency chaining on DNS validation rather than the certificate creation

### PR DESCRIPTION
**Pull request by tommij**
Given resource dependencies on an ACM certificate:
`
module "foo" {
    source              = "github.com/traveloka/terraform-aws-acm-certificate"
  domain_name         = "host.some.fqdn"
  hosted_zone_name    = "some.fqdn"
  is_hosted_zone_private = "false"
  validation_method   = "DNS"
  certificate_name    = "host-some-fqdn"
  environment         = "test"
  description         = "TLS certificate for host.some.fqdn"
  product_domain      = "product-name"
  providers {
    aws = "aws.us-east-1"
  }
}

resource "aws_some_resource_that_needs_acm" "foo" {
  foo = "foo"
  bar = "bar"
  certificate_arn = "${module.foo.acm_certificate_arn}|
}
`
Dependency fails, because the certificate is not validated, and thus not issued.
This PR returns the ACM certificate arn from the validator if the validation is DNS, allowing dependencies to be fulfilled.

### CODE REVIEW

**Comment by tommij** [comment]
Hey @SiahaanBernard 

I'm entirely fine with removing Email, I did the ternary to maintain support for it.
However:

Neither of the two current outputs are useful for dependency chaining (which is why I modified the ACM output to be aws_acm_certificate_validation.dns_validation.*.certificate_arn[0] )


As-is, you can easily have a non-validated ACM pass on to the next steps in a manifest, which will cause them to fail.

I would suggest to modify acm_certificate_arn output to: 
```aws_acm_certificate_validation.dns_validation.*.certificate_arn[0]```
Or at least have a third output, e.g. acm_validated_certificate_arn using that resource.

TL;DR
Just because you have a certificate arn from aws_acm_certificate, and a route53 validation record does not mean it's validated, and resources depending on the certificate arn will likely fail when you use them as if it was validated.
